include/zephyr/lw_sched/lw_sched.h

```diff
@@ -0,0 +1,221 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ */
+
+#ifndef __SUBSYS_LIGHTWEIGHT_SCHEDULER
+#define __SUBSYS_LIGHTWEIGHT_SCHEDULER
+
+#include <zephyr/kernel.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Lightweight Scheduler
+ * @defgroup subsys_lw_sched Lightweight Scheduler
+ * @ingroup os_services
+ * @{
+ */
+
+struct lw_scheduler;
+struct lw_task;
+
+/** @brief Lightweight task operation handlers */
+struct lw_task_ops {
+	int  (*execute)(void *arg);   /**< Execute handler */
+	void (*abort)(void *arg);     /**< Optional abort handler */
+};
+
+/** @brief Arguments to lightweight task operation handlers */
+struct lw_task_args {
+	void *execute;                /**< Args for execute handler */
+	void *abort;                  /**< Args for abort handler */
+};
+
+#define LW_TASK_PAUSED  0             /**< Task is in a paused state */
+#define LW_TASK_EXECUTE 1             /**< Task is executing */
+#define LW_TASK_ABORT   2             /**< Task is aborted */
+
+/** @brief Lightweight task structure */
+struct lw_task {
+	sys_dnode_t node;               /**< Node in list of tasks */
+	sys_dnode_t reorder;            /**< Node in reorder list */
+	struct lw_scheduler *sched;     /**< Associated scheduler */
+	const struct lw_task_ops *ops;  /**< Task operations */
+	struct lw_task_args *args;      /**< Task operation arguments */
+	uint32_t state;                 /**< Task state */
+	uint32_t order;                 /**< Task order */
+	uint32_t new_order;             /**< New task order */
+	uint32_t delay;                 /**< # of intervals to delay */
```

Comment:
needs to be escaped (it's picked up as a H1 heading otherwise), but feel free to spell it "Number" if that looks prettier to you :)

```suggestion
	uint32_t delay;                 /**< \# of intervals to delay */
```

Reply:
Fixing.